debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnectorTask.java

```diff
@@ -287,25 +287,27 @@ private void validate(MongoDbConnection mongoDbConnection, Offsets<MongoDbPartit
                 return;
             }
 
-            boolean logPositionAvailable = mongoDbConnection.validateLogPosition(offset, taskContext);
+            if (connectorConfig.isLogPositionCheckEnabled()) {
+                boolean logPositionAvailable = mongoDbConnection.validateLogPosition(offset, taskContext);
 
-            if (!logPositionAvailable) {
-                LOGGER.warn("Last recorded offset is no longer available on the server.");
+                if (!logPositionAvailable && !offset.isSnapshotRunning()) {
```

Comment:
Is `&& !offset.isSnapshotRunning()` required? It is already checked on the previous `if`